code/test/util.js

```diff
@@ -257,7 +427,234 @@ const printInnerTrees = ({ Debugger, innerTrees }) => {
+      identificationKeys.push(x[0].toString())
+    }
+  } catch (ex) {
+    console.log(`Failed to parse walletIdentificationKeys: ${ex.toString()}`)
+  }
+  const forwardAddress = (await wallet.getForwardAddress()).toString()
+  const walletInfo = await wallet.getInfo()
+  let info = {}
+  try {
+    info = {
+      root: walletInfo[0].toString(),
+      height: new BN(walletInfo[1]).toNumber(),
+      interval: new BN(walletInfo[2]).toNumber(),
+      t0: new BN(walletInfo[3]).toNumber(),
+      lifespan: new BN(walletInfo[4]).toNumber(),
+      maxOperationsPerInterval: new BN(walletInfo[5]).toNumber(),
+      recoveryAddress: walletInfo[6].toString(),
+      extra: new BN(walletInfo[7]).toNumber()
+    }
+  } catch (ex) {
+    console.log(`Failed to parse Wallet Info: ${ex.toString()}`)
+  }
+  const walletOldInfo = await wallet.getOldInfos()
+  let oldInfo = []
+  try {
+    for (let x of walletOldInfo) {
+      oldInfo.push({
+        root: x[0].toString(),
+        height: new BN(x[1]).toNumber(),
+        interval: new BN(x[2]).toNumber(),
+        t0: new BN(x[3]).toNumber(),
+        lifespan: new BN(x[4]).toNumber(),
+        maxOperationsPerInterval: new BN(x[5]).toNumber()
+      })
+    }
+  } catch (ex) {
+    console.log(`Failed to parse Old Info: ${ex.toString()}`)
+  }
+  const walletInnerCores = await wallet.getInnerCores()
+  let innerCores = []
+  try {
+    for (let x of walletInnerCores) {
+      innerCores.push({
+        root: x[0].toString(),
+        height: new BN(x[1]).toNumber(),
+        interval: new BN(x[2]).toNumber(),
+        t0: new BN(x[3]).toNumber(),
+        lifespan: new BN(x[4]).toNumber(),
+        maxOperationsPerInterval: new BN(x[5]).toNumber()
+      })
+    }
+  } catch (ex) {
+    console.log(`Failed to parse walletInnerCores: ${ex.toString()}`)
+  }
+  const rootKey = (await wallet.getRootKey()).toString()
+  const walletVersion = await wallet.getVersion()
+  let version = {}
+  try {
+    version = {
+      majorVersion: new BN(walletVersion[0]).toNumber(),
+      minorVersion: new BN(walletVersion[1]).toNumber()
+    }
+  } catch (ex) {
+    console.log(`Failed to parse walletVersion: ${ex.toString()}`)
+  }
+  const walletSpendingState = await wallet.getSpendingState()
+  let spendingState = {}
+  try {
+    spendingState = {
+      spendingLimit: walletSpendingState[0].toString(),
+      spentAmount: walletSpendingState[1].toString(),
+      lastSpendingInterval: walletSpendingState[2].toString(),
+      spendingInterval: walletSpendingState[3].toString(),
+      lastLimitAdjustmentTime: walletSpendingState[4].toString(),
+      highestSpendingLimit: walletSpendingState[5].toString()
+    }
+  } catch (ex) {
+    console.log(`Failed to parse walletSpendingState: ${ex.toString()}`)
+  }
+  const nonce = new BN(await wallet.getNonce()).toNumber()
+  const lastOperationTime = new BN(await wallet.lastOperationTime()).toNumber()
+  const walletAllCommits = await wallet.getAllCommits()
```

Comment:
```
const [hashes, paramsHashes, verificationHashes, timestamps, completed] = Object.keys(walletAllCommits).map(k => walletAllCommits[k])
const commits = hashes.map((e, i) => ({ hash: hashes[i], paramsHash: paramsHashes[i], verificationHash: verificationHashes[i], timestamp: timestamps[i], completed: completed[i] })
```

Reply:
can be used for all structure below

Reply:
can be used for all structure below

Reply:
Looks very useful. 🙏 
I added in the code and tested and got the error below.
So leaving it in as comments until the error is resolved.

```
     AssertionError: wallet.allCommits is incorrect: expected [ Array(1) ] to deeply equal { Object (0, 1, ...) }
      at Object.checkONEWalletStateChange (test/util.js:615:10)
      at Context.<anonymous> (test/general.js:280:20)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

Reply:
that is saying you get two forms of representations (array of objects with named properties, v.s. just a bunch of arrays) , you need to stick with one form everywhere